### ISSUE TITLE
Discovery not working with Migic 2 LAN

### ISSUE THREAD

**Issue by MassiAtGithub** [2Fake/devolo_plc_api#39]
**Describe the bug**
Used the autodiscovery feature from the library (sync method: discover_network) but I didn't receive any answer from the two Devolo Magic 2 LAN devices.
I've tryied to rev-eng the working behaviour of the desktop original software and I've found a difference on the seq of udp packets.
In details:
1) the 'devolo cockpit' starts asking a GET_SW.REQ using Homeplug AV on broadcast - I guess it would not work because Magic2 doesn't use Homeplug standard (img #1).
2) then the cockpit send a broadcast req on port 19375/udp with message 'whoisthere' and the source IP - this is probably porprietary so I have no idea of how it works (img #2).
3) later it uses mDNS and asks for 3 records in the same query: _dvl_deviceapi._tcp.local, _dvl-plcnetapi._tcp.local and _http._tcp.local. Req is QM (unicast disabled - see img #3). This is the point where the 'discover_network' starts to overlap.
4) finally I saw the 2 devices sending unicast UDP responses to the cockpit using the same ports defined at point #2, src/dst inverted. The PDU is rich of data, see see img #4.

Using the lib I dont get the last step so i guess there is something wrong/missing somewhere.

**Used device**
Devolo Maginc LAN 2 1-1

Cheers and thanks!

![4](https://user-images.githubusercontent.com/82322604/114311686-06180c80-9af0-11eb-9a7f-a1ddec151f59.png)
![3](https://user-images.githubusercontent.com/82322604/114311688-06180c80-9af0-11eb-83c6-969d92e76fab.png)
![1](https://user-images.githubusercontent.com/82322604/114311697-0c0ded80-9af0-11eb-9021-a393a90babc8.png)
![2](https://user-images.githubusercontent.com/82322604/114311698-0ca68400-9af0-11eb-9d86-784e27b6de2b.png)

**Comment by Shutgun** [comment]
Hi @MassiAtGithub,

the behaviour indeed changed. In newer firmware versions the Magic LAN devices dropped mDNS support. I'll try to figure out what to do with them.

**Comment by eikel** [comment]
Thanks to the information given here, I have built a similar `whoisthere` UDP packet using `socat` that makes my two devolo Magic 2 LAN adapters (firmware 7.10.2.77 (2021-04-01)) respond with a `hello` UDP packet.

I observed that setting the last number in the `whoisthere` packet to 1 makes the adapters respond directly to the sender's IP address. Using another number (e.g., 2), makes them respond with an UDP broadcast to port 19375.

Start an UDP receiver:
```sh
socat -u UDP4-RECVFROM:19375,fork system:'cat'
```

Sending the `whoisthere` UDP broadcast packet:
```sh
echo -n "whoisthere\x00192.168.1.255\x00255.255.255.0\x00\x002\x00" | socat - UDP4-DATAGRAM:192.168.1.255:19375,broadcast
```

The response contains an XML document (formatted for better readability and censored for my privacy):
```xml
<?xml version="1.0"?>
<status>
  <device-type>Magic 2 LAN triple</device-type>
  <hostname>devolo-hostname</hostname>
  <mt-number>4321</mt-number>
  <serial-number>1234567890123456</serial-number>
  <deviceapi>
    <features>led,reset,intmtg1,update</features>
    <path>Keeh3shozeijezou/deviceapi</path>
    <port>14791</port>
    <version>v0</version>
  </deviceapi>
  <firmware>
    <date>2021-04-01</date>
    <version>7.10.2.77</version>
  </firmware>
  <plcnetapi>
    <path>ougie4ohvaeBahfi/plcnetapi</path>
    <port>14791</port>
    <version>v0</version>
  </plcnetapi>
  <powerline>
    <mac-address>aa:bb:cc:dd:ee:ff</mac-address>
    <technology>ghn</technology>
  </powerline>
</status>
```

**Comment by eikel** [comment]
I found some information about that protocol on UDP port 19375 [in another Python project](https://github.com/activecm/passer/blob/d59b1824b889a80e38235c17d7eb139cfeb76f36/passer.py#L1479). Especially the [first link](https://flambda.de/2013/06/18/audioextender/) there contains some code.

**Comment by Shutgun** [comment]
Hi @eikel , from a trusty source I know, that mDNS will be back in the LAN devices soon. Maybe we should wait a few weeks and see, if that will work as expected.

**Comment by Shutgun** [comment]
Hi @eikel , hi @MassiAtGithub , I just released [0.6.0](https://github.com/2Fake/devolo_plc_api/releases/tag/v0.6.0) which should fix your issues as long as the firmware of the device is up-to-date.
